### PR TITLE
Conf: Added Setup Framework

### PR DESCRIPTION
**Pull request by ujjwalpl**
Signed-off-by: Ujjwal Lanjewar <ujjwal.lanjewar@seagate.com>

### CODE REVIEW

File: py-utils/src/utils/setup/utils/setup_cli.py [hunk 1]
```diff
@@ -87,6 +87,17 @@ def process(self):
...
 
+class PrepareCmd(Cmd):
+    """ Prepare Setup Cmd """
+    name = "prepare"
+
+    def __init__(self, args: dict):
+        super().__init__(args)
+
+    def process(self):
+        pass
```
Comment (sachinpunadikar) [inline py-utils/src/utils/setup/utils/setup_cli.py:98]: Need to return 0 to indicate a success.
If "pass" is taking care of that , then no change is needed

File: py-utils/src/utils/setup/kafka/kafka_setup.py [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+
+class PostInstallCmd(Cmd):
+    """ PostInstall Setup Cmd """
+    name = "post_install"
+
+    def __init__(self, args: dict):
+        super().__init__(args)
+
+    def process(self):
+        pass
```
Comment (sachinpunadikar) [inline py-utils/src/utils/setup/kafka/kafka_setup.py:85]: Need to return 0 to indicate success. If "pass" is capable of doing this, then ignore this comment.
Take care of all the commands likewise.

File: py-utils/src/utils/setup/kafka/kafka.py [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+    def __init__(self, rc, message, *args):
+        self._rc = rc
+        self._desc = message % (args)
+
+    def __str__(self):
+        if self._rc == 0: return self._desc
+        return "error(%d): %s" %(self._rc, self._desc)
+
+
+class Utils:
```
Comment (sachinpunadikar) [inline py-utils/src/utils/setup/kafka/kafka.py:35]: I suppose the class name should be Kafka

**Review by sachinpunadikar — CHANGES_REQUESTED** [review]
Please check the comments and make changes accordingly

File: py-utils/setup.py [hunk 4]
```diff
@@ -78,12 +78,13 @@ def get_install_requirements() -> list:
...
             'conf = cortx.utils.conf_store.conf_cli:main',
-            'setup = cortx.utils.setup.setup_cli:main'
+            'utils_setup = cortx.utils.setup.utils.setup_cli:main'
+            'kafka_setup = cortx.utils.setup.kafka.kafka_setup:main'
         ]
       },
       data_files = [ ('/var/lib/cortx/ha/specs', specs),
                      ('/var/lib/cortx/ha', ['src/utils/ha/hac/args.yaml', 'src/utils/ha/hac/re_build.sh']),
-                     ('/opt/seagate/cortx/utils/conf', ['requirements.txt', 'src/utils/setup/setup.yaml'])],
+                     ('/opt/seagate/cortx/utils/conf', ['requirements.txt', 'src/utils/setup/utils/setup.yaml'])],
```
Comment (andkononykhin2) [inline py-utils/setup.py:87]: @ujjwalpl @sachinpunadikar It's not a change of the current PR but I've just noticed that you install some data files into global locations. Is root-based installation considered only? Would it be possible to use cortx-utils in non-root python isolated environment as it usually happens for python projects?

Comment (sachinpunadikar) [reply py-utils/setup.py:87]: @andkononykhin2, currently we are considering root-based installations. But in future we can work on non-root based environment.

File: py-utils/src/utils/setup/utils/setup_cli.py (hunk 1 above)
Comment (ujjwalpl) [reply py-utils/src/utils/setup/utils/setup_cli.py:98]: Done

File: py-utils/src/utils/setup/kafka/kafka_setup.py (hunk 2 above)
Comment (ujjwalpl) [reply py-utils/src/utils/setup/kafka/kafka_setup.py:85]: Done

File: py-utils/src/utils/setup/kafka/kafka.py (hunk 3 above)
Comment (ujjwalpl) [reply py-utils/src/utils/setup/kafka/kafka.py:35]: Done

**Review by sachinpunadikar — APPROVED** [review]
+1

**Comment by sachinpunadikar** [comment]
Please work on the conflicting files.

**Review by rahul-tripathi-git — APPROVED** [review]
+1

**Comment by sachinpunadikar** [comment]
RPM installation is failing. Need to fix this. I did mentioned this earlier as well .. post_install script will need change.
23:24:41      Running transaction
23:24:41        Installing : cortx-py-utils-2.0.0-174_fb22b99.noarch                      1/1
23:24:41      /bin/cp: cannot stat '/usr/lib/python3.6/site-packages/cortx/utils/setup/setup_cli.py': No such file or directory
23:24:41      /bin/chmod: cannot access '/opt/seagate/cortx/utils/bin/utils_setup': No such file or directory
23:24:41      warning: %post(cortx-py-utils-2.0.0-174_fb22b99.noarch) scriptlet failed, exit status 1

**Review by s-arya — APPROVED** [review]

**Pull request merged**